Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -19,11 +19,52 @@ let headerSeparator: StaticString = ": "
 
 /// A representation of the request line and header fields of a HTTP request.
 public struct HTTPRequestHead: Equatable {
+    private class _Storage {
+        var method: HTTPMethod
+        var rawURI: URI
+        var version: HTTPVersion
+
+        init(method: HTTPMethod, rawURI: URI, version: HTTPVersion) {
+            self.method = method
+            self.rawURI = rawURI
+            self.version = version
+        }
+
+        func copy() -> _Storage {
+            return _Storage(method: self.method, rawURI: self.rawURI, version: self.version)
```

Comment:
Can you either always use `.init` or always `_Storage` just to make it consistent ?